### PR TITLE
Fix devserver unexpected error on incorrect config & add warnings

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

The dev server should not use `invariant` to check if the server already exists on buildFailure as sometimes the server will not have started because of an incorrect config and throw the wrong error in that case.

Also changes all other invariants to be warnings as we probably don't want to throw exceptions if it happens as people should still be able to use parcel if it happens.

Related #3960

## 💻 Examples

See  #3960

### CODE REVIEW

File: packages/reporters/dev-server/src/ServerReporter.js [hunk 1]
```diff
@@ -51,7 +54,12 @@ export default new Reporter({
         break;
       }
       case 'watchEnd':
-        invariant(server != null);
+        if (!server) {
+          return logger.warn({
+            message: 'Could not shutdown devserver because it does not exist.',
```
Comment (DeMoorJasper) [inline packages/reporters/dev-server/src/ServerReporter.js]: This should actually never happen, but I guess better to be safe

File: packages/reporters/dev-server/src/ServerReporter.js [hunk 2]
```diff
@@ -60,14 +68,23 @@ export default new Reporter({
         hmrServers.delete(serve.port);
         break;
       case 'buildSuccess':
-        invariant(server != null);
+        if (!server) {
+          return logger.warn({
+            message:
+              'Could not send success event to devserver because it does not exist.',
```
Comment (DeMoorJasper) [inline packages/reporters/dev-server/src/ServerReporter.js]: Same as other comment, should in theory never happen

File: packages/reporters/dev-server/src/ServerReporter.js [hunk 3]
```diff
@@ -19,7 +18,11 @@ export default new Reporter({
       case 'watchStart': {
         // If there's already a server when watching has just started, something
         // is wrong.
-        invariant(server == null);
+        if (server) {
+          return logger.warn({
+            message: 'Trying to create the devserver but it already exists.',
```
Comment (DeMoorJasper) [inline packages/reporters/dev-server/src/ServerReporter.js]: Same as other comment, should in theory never happen

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>packages/benchmarks/kitchen-sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>packages/benchmarks/react-hn ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.35s | -361.00ms |
| Cached | 3.73s | -46.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/module/index.js | 34.33kb | +0.00b | 72.00ms | -234.00ms 🚀 |
| dist/main/index.js | 34.08kb | +0.00b | 77.00ms | -234.00ms 🚀 |

</p></details>

<details><summary>packages/benchmarks/ak-editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 3.15m | -1.34s |
| Cached | 6.18s | -170.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/pdfRenderer.85e56a51.js | 1.11mb | +0.00b | 36.72s | -2.32s 🚀 |
| dist/EmojiPickerComponent.1003396c.js | 100.51kb | +0.00b | 43.12s | -2.77s 🚀 |
| dist/Toolbar.1699663e.js | 94.13kb | +0.00b | 1.23m | +13.48s ⚠️ |
| dist/component.10b390dc.js | 42.89kb | +0.00b | 38.99s | -2.40s 🚀 |
| dist/popup.b8e50b5b.js | 20.67kb | +0.00b | 38.25s | -2.33s 🚀 |
| dist/popup.8dca552f.js | 20.13kb | +0.00b | 38.25s | -2.33s 🚀 |
| dist/popup.8d8f40ed.js | 19.10kb | +0.00b | 38.25s | -2.33s 🚀 |
| dist/EmojiPickerComponent.bcce358c.js | 17.29kb | +0.00b | 42.06s | -2.67s 🚀 |
| dist/popup.26e30eb7.js | 12.04kb | +0.00b | 38.25s | -2.33s 🚀 |
| dist/workerHasher.d7e20c98.js | 11.57kb | +0.00b | 44.06s | -2.57s 🚀 |
| dist/EmojiTypeAheadComponent.1c09c30d.js | 9.33kb | +0.00b | 43.77s | -2.55s 🚀 |
| dist/card.f99e1597.js | 5.87kb | +0.00b | 13.69s | -1.28s 🚀 |
| dist/component.a26e593f.js | 4.13kb | +0.00b | 39.20s | -2.44s 🚀 |
| dist/component.8fa44754.js | 3.38kb | +0.00b | 39.20s | -2.44s 🚀 |
| dist/EmojiUploadComponent.0460cc59.js | 2.30kb | +0.00b | 43.77s | -2.55s 🚀 |
| dist/component.ab65829f.js | 2.30kb | +0.00b | 39.20s | -2.44s 🚀 |
| dist/16.dcacbe5c.js | 1.69kb | +0.00b | 39.61s | -2.38s 🚀 |
| dist/media-picker-analytics-error-boundary.0f11d632.js | 1.64kb | +0.00b | 38.99s | -2.40s 🚀 |
| dist/date.cfe6153f.js | 1.63kb | +0.00b | 39.77s | -2.39s 🚀 |
| dist/16.de3f3184.js | 1.63kb | +0.00b | 39.34s | -2.41s 🚀 |
| dist/images.c60a97a9.js | 1.57kb | +0.00b | 39.79s | -2.39s 🚀 |
| dist/feedback.01d7bcf7.js | 1.53kb | +0.00b | 42.06s | -2.35s 🚀 |
| dist/card.2c902906.js | 1.52kb | +0.00b | 13.71s | -1.26s 🚀 |
| dist/workerHasher.854262ce.js | 1.49kb | +0.00b | 43.77s | -2.85s 🚀 |
| dist/status.e3ed4764.js | 1.35kb | +0.00b | 40.01s | -2.44s 🚀 |
| dist/list-number.dbe744df.js | 1.35kb | +0.00b | 39.84s | -2.45s 🚀 |
| dist/16.f54dac28.js | 1.34kb | +0.00b | 39.46s | -2.43s 🚀 |
| dist/16.8f0e8fe1.js | 1.29kb | +0.00b | 39.34s | -2.41s 🚀 |
| dist/16.43b62c01.js | 1.29kb | +0.00b | 39.35s | -2.44s 🚀 |
| dist/code.092e0137.js | 1.28kb | +0.00b | 39.71s | -2.38s 🚀 |
| dist/16.b17a4d30.js | 1.27kb | +0.00b | 39.70s | -2.37s 🚀 |
| dist/16.51bbc68e.js | 1.24kb | +0.00b | 39.63s | -2.39s 🚀 |
| dist/link.6fa27bb2.js | 1.20kb | +0.00b | 39.84s | -2.42s 🚀 |
| dist/heading6.3996fe48.js | 1.19kb | +0.00b | 40.07s | -2.47s 🚀 |
| dist/16.385d3ff5.js | 1.19kb | +0.00b | 39.61s | -2.38s 🚀 |
| dist/heading3.9fea5413.js | 1.18kb | +0.00b | 40.06s | -2.47s 🚀 |
| dist/16.b3ce4050.js | 1.16kb | +0.00b | 39.34s | -2.45s 🚀 |
| dist/16.e37bbe6b.js | 1.15kb | +0.00b | 39.61s | -2.40s 🚀 |
| dist/16.f7fe274a.js | 1.14kb | +0.00b | 39.70s | -2.38s 🚀 |
| dist/emoji.c4bc5355.js | 1.12kb | +0.00b | 39.77s | -2.41s 🚀 |
| dist/16.ac2ff83f.js | 1.10kb | +0.00b | 39.48s | -2.42s 🚀 |
| dist/16.4c652b29.js | 1.10kb | +0.00b | 39.46s | -2.42s 🚀 |
| dist/16.24165259.js | 1.09kb | +0.00b | 39.34s | -2.45s 🚀 |
| dist/16.74510c56.js | 1.09kb | +0.00b | 39.46s | -2.41s 🚀 |
| dist/16.97c5bdf9.js | 1.09kb | +0.00b | 39.46s | -2.44s 🚀 |
| dist/16.a79db876.js | 1.09kb | +0.00b | 39.22s | -2.42s 🚀 |
| dist/heading5.2f045997.js | 1.07kb | +0.00b | 40.06s | -2.48s 🚀 |
| dist/16.6f8ac62a.js | 1.05kb | +0.00b | 39.20s | -2.44s 🚀 |
| dist/expand.13380081.js | 1.05kb | +0.00b | 42.06s | -2.35s 🚀 |
| dist/card.aae6ebb9.js | 1.04kb | +0.00b | 13.69s | -1.28s 🚀 |
| dist/16.3e36da0b.js | 1.01kb | +0.00b | 39.61s | -2.40s 🚀 |
| dist/heading2.2194210a.js | 1.00kb | +0.00b | 40.06s | -2.47s 🚀 |
| dist/component.8351f228.js | 999.00b | +0.00b | 38.99s | -2.40s 🚀 |
| dist/mention.9fdad24e.js | 983.00b | +0.00b | 39.85s | -2.44s 🚀 |
| dist/fallback.305c8864.js | 982.00b | +0.00b | 40.01s | -2.45s 🚀 |
| dist/heading4.63d1ca06.js | 978.00b | +0.00b | 40.06s | -2.48s 🚀 |
| dist/layout.6f66c2da.js | 964.00b | +0.00b | 39.84s | -2.42s 🚀 |
| dist/divider.160f3108.js | 942.00b | +0.00b | 39.77s | -2.41s 🚀 |
| dist/quote.88bfdca4.js | 938.00b | +0.00b | 40.01s | -2.44s 🚀 |
| dist/action.146450ba.js | 921.00b | +0.00b | 39.70s | -2.38s 🚀 |
| dist/decision.06254b93.js | 900.00b | +0.00b | 39.77s | -2.39s 🚀 |
| dist/panel-warning.59c8ef3f.js | 893.00b | +0.00b | 39.93s | -2.44s 🚀 |
| dist/list.4f623de3.js | 872.00b | +0.00b | 39.84s | -2.45s 🚀 |
| dist/heading1.65fb8482.js | 866.00b | +0.00b | 40.02s | -2.44s 🚀 |
| dist/panel-error.c2479698.js | 793.00b | +0.00b | 39.93s | -2.42s 🚀 |
| dist/panel.23cd7224.js | 790.00b | +0.00b | 39.94s | -2.44s 🚀 |
| dist/table.4ace5843.js | 782.00b | +0.00b | 40.01s | -2.45s 🚀 |
| dist/panel-success.6a7b97d7.js | 730.00b | +0.00b | 39.93s | -2.45s 🚀 |
| dist/panel-note.5ff85bdc.js | 729.00b | +0.00b | 39.93s | -2.42s 🚀 |
| dist/media-viewer-analytics-error-boundary.dd752a9e.js | 725.00b | +0.00b | 13.69s | -1.28s 🚀 |
| dist/simpleHasher.884d4eb6.js | 450.00b | +0.00b | 44.06s | -2.57s 🚀 |
| dist/dropzone.976debcc.js | 179.00b | +1.00b ⚠️ | 1.14m | +27.95s ⚠️ |
| dist/clipboard.976debcc.js | 179.00b | +1.00b ⚠️ | 38.99s | -2.40s 🚀 |
| dist/browser.976debcc.js | 179.00b | +1.00b ⚠️ | 59.38s | +17.99s ⚠️ |
| dist/editorView.976debcc.js | 179.00b | -1.00b 🚀 | 38.25s | -22.17s 🚀 |
| dist/media-card-analytics-error-boundary.976debcc.js | 178.00b | -5.00b 🚀 | 39.70s | -30.59s 🚀 |
| dist/16.976debcc.js | 176.00b | -5.00b 🚀 | 38.99s | -3.08s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/smartMediaEditor.678dca17.js | 591.21kb | +0.00b | 140.00ms | +18.00ms ⚠️ |
| dist/EmojiPickerComponent.1003396c.js | 100.52kb | +0.00b | 135.00ms | +16.00ms ⚠️ |
| dist/Toolbar.1699663e.js | 94.14kb | +0.00b | 130.00ms | +16.00ms ⚠️ |
| dist/ui.c85d9c1b.js | 56.40kb | +0.00b | 130.00ms | +17.00ms ⚠️ |
| dist/smartMediaEditor.25b0ed57.js | 55.87kb | +0.00b | 129.00ms | +16.00ms ⚠️ |
| dist/DatePicker.0242ec52.js | 22.09kb | +0.00b | 132.00ms | +16.00ms ⚠️ |
| dist/EmojiPickerComponent.bcce358c.js | 17.29kb | +0.00b | 132.00ms | +16.00ms ⚠️ |
| dist/workerHasher.d7e20c98.js | 11.57kb | +0.00b | 131.00ms | +17.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.1c09c30d.js | 9.33kb | +0.00b | 131.00ms | +16.00ms ⚠️ |
| dist/index.css | 3.38kb | +0.00b | 119.00ms | +17.00ms ⚠️ |
| dist/EmojiUploadComponent.0460cc59.js | 2.30kb | +0.00b | 132.00ms | +17.00ms ⚠️ |
| dist/date.cfe6153f.js | 1.63kb | +0.00b | 144.00ms | +18.00ms ⚠️ |
| dist/images.c60a97a9.js | 1.57kb | +0.00b | 142.00ms | +17.00ms ⚠️ |
| dist/feedback.01d7bcf7.js | 1.53kb | +0.00b | 134.00ms | +17.00ms ⚠️ |
| dist/workerHasher.854262ce.js | 1.49kb | +0.00b | 131.00ms | +17.00ms ⚠️ |
| dist/status.e3ed4764.js | 1.35kb | +0.00b | 137.00ms | +17.00ms ⚠️ |
| dist/list-number.dbe744df.js | 1.35kb | +0.00b | 141.00ms | +18.00ms ⚠️ |
| dist/code.092e0137.js | 1.28kb | +0.00b | 144.00ms | +17.00ms ⚠️ |
| dist/16.b17a4d30.js | 1.27kb | +0.00b | 145.00ms | +16.00ms ⚠️ |
| dist/link.6fa27bb2.js | 1.20kb | +0.00b | 141.00ms | +17.00ms ⚠️ |
| dist/heading6.3996fe48.js | 1.19kb | +0.00b | 134.00ms | +16.00ms ⚠️ |
| dist/heading3.9fea5413.js | 1.18kb | +0.00b | 135.00ms | +16.00ms ⚠️ |
| dist/16.f7fe274a.js | 1.14kb | +0.00b | 145.00ms | +17.00ms ⚠️ |
| dist/emoji.c4bc5355.js | 1.12kb | +0.00b | 145.00ms | +17.00ms ⚠️ |
| dist/heading5.2f045997.js | 1.07kb | +0.00b | 135.00ms | +17.00ms ⚠️ |
| dist/expand.13380081.js | 1.05kb | +0.00b | 133.00ms | +16.00ms ⚠️ |
| dist/heading2.2194210a.js | 1.00kb | +0.00b | 136.00ms | +17.00ms ⚠️ |
| dist/mention.9fdad24e.js | 983.00b | +0.00b | 140.00ms | +18.00ms ⚠️ |
| dist/fallback.305c8864.js | 982.00b | +0.00b | 135.00ms | +16.00ms ⚠️ |
| dist/heading4.63d1ca06.js | 978.00b | +0.00b | 135.00ms | +17.00ms ⚠️ |
| dist/layout.6f66c2da.js | 964.00b | +0.00b | 141.00ms | +17.00ms ⚠️ |
| dist/divider.160f3108.js | 942.00b | +0.00b | 146.00ms | +18.00ms ⚠️ |
| dist/quote.88bfdca4.js | 938.00b | +0.00b | 137.00ms | +17.00ms ⚠️ |
| dist/action.146450ba.js | 921.00b | +0.00b | 144.00ms | +17.00ms ⚠️ |
| dist/decision.06254b93.js | 900.00b | +0.00b | 143.00ms | +17.00ms ⚠️ |
| dist/panel-warning.59c8ef3f.js | 893.00b | +0.00b | 138.00ms | +17.00ms ⚠️ |
| dist/list.4f623de3.js | 872.00b | +0.00b | 140.00ms | +17.00ms ⚠️ |
| dist/heading1.65fb8482.js | 866.00b | +0.00b | 136.00ms | +16.00ms ⚠️ |
| dist/panel-error.c2479698.js | 793.00b | +0.00b | 140.00ms | +18.00ms ⚠️ |
| dist/panel.23cd7224.js | 790.00b | +0.00b | 138.00ms | +17.00ms ⚠️ |
| dist/table.4ace5843.js | 782.00b | +0.00b | 136.00ms | +17.00ms ⚠️ |
| dist/panel-success.6a7b97d7.js | 730.00b | +0.00b | 139.00ms | +18.00ms ⚠️ |
| dist/panel-note.5ff85bdc.js | 729.00b | +0.00b | 139.00ms | +17.00ms ⚠️ |
| dist/simpleHasher.884d4eb6.js | 450.00b | +0.00b | 130.00ms | +17.00ms ⚠️ |
| dist/esm.976debcc.js | 178.00b | +2.00b ⚠️ | 161.00ms | -0.00ms |
| dist/media-card-analytics-error-boundary.976debcc.js | 178.00b | +2.00b ⚠️ | 158.00ms | -0.00ms |
| dist/dropzone.976debcc.js | 178.00b | +2.00b ⚠️ | 156.00ms | +1.00ms |
| dist/clipboard.976debcc.js | 178.00b | +2.00b ⚠️ | 155.00ms | -0.00ms |
| dist/browser.976debcc.js | 178.00b | +2.00b ⚠️ | 155.00ms | +1.00ms |
| dist/16.976debcc.js | 178.00b | +2.00b ⚠️ | 145.00ms | +17.00ms ⚠️ |
| dist/editorView.976debcc.js | 178.00b | +2.00b ⚠️ | 128.00ms | +16.00ms ⚠️ |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/256644977001497089)**

**Pull request merged**